services/galley/src/Galley/API/Query.hs

```diff
@@ -115,6 +121,30 @@ getRemoteConversations zusr remoteConvs = do
     gcresConvs <$> runFederatedGalley remoteDomain rpc
   pure $ concat convs
 
+getRemoteConversationsWithFailures :: UserId -> [Remote ConvId] -> Galley ([Qualified ConvId], [Public.Conversation])
+getRemoteConversationsWithFailures zusr remoteConvs = do
+  localDomain <- viewFederationDomain
+  let qualifiedZUser = Qualified zusr localDomain
+  let convsByDomain = partitionRemote remoteConvs
+  convs <- pooledForConcurrentlyN 8 convsByDomain $ \(remoteDomain, convIds) -> handleFailures remoteDomain convIds $ do
+    let req = FederatedGalley.GetConversationsRequest qualifiedZUser convIds
+        rpc = FederatedGalley.getConversations FederatedGalley.clientRoutes req
+    gcresConvs <$> executeFederated remoteDomain rpc
+  pure $ concatEithers convs
+  where
+    handleFailures :: Domain -> [ConvId] -> ExceptT FederationError Galley a -> Galley (Either [Qualified ConvId] a)
```

Comment:
Is my assumption correct that this `handleFailures` function is not tested? Could we make this a top-level function and exercise a failure case in a test here? (Or maybe add a FUTUREWORK for this).

Reply:
It is exercised from integration tests when  Alice tries to get a remote conv on c.far-away.example.com, for which the federated call fails.

Reply:
ok cool, I had not seen that.